File: Treasury.Application/Contracts/V1/Requests/GeneralPagedRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Treasury.Application.Contracts.V1.Requests;

public class GeneralPagedRequest : IPagedRequest
{
    public string[] Name { get; set; } = Array.Empty<string>();

    public string[] Acronym { get; set; } = Array.Empty<string>();

    public string[] Classification { get; set; } = Array.Empty<string>();

    public string[] Type { get; set; } = Array.Empty<string>();

    public bool IncludeInactive { get; set; } = false;

    [Required]
    public int Page { get; set; } = 1;

    [Required]
    public int Rpp { get; set; } = 10;
}

File: src/main/java/edu/wpi/sga/treasury/api/contract/request/GeneralPagedRequest.java
package edu.wpi.sga.treasury.api.contract.request;


import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class GeneralPagedRequest implements PagedRequest {
    private Integer page = 1;
    private Integer resultsPerPage = 10;

    private List<String> name = List.of();
    private List<String> acronym = List.of();
    private List<String> classification = List.of();
    private List<String> type = List.of();
    private boolean includeInactive = false;
}
